src/test/java/co/rsk/federate/signing/hsm/message/AdvanceBlockchainMessageTest.java

```diff
@@ -82,4 +84,99 @@ public void test_getParsedBrothers_invalid_blockHeader() throws HSMBlockchainBoo
+        List<String> parsedBlockHeaders = message.getParsedBlockHeaders();
+
+        String[] blockBrothers1 = message.getParsedBrothers(parsedBlockHeaders.get(1));
+        for (int i = 0; i < blockBrothers1.length - 1; ++i) {
+            assertTrue(blockBrothers1[i].compareTo(blockBrothers1[i + 1]) <= 0);
+        }
+
+        String[] blockBrothers2 = message.getParsedBrothers(parsedBlockHeaders.get(0));
+        for (int i = 0; i < blockBrothers2.length - 1; ++i) {
+            assertTrue(blockBrothers2[i].compareTo(blockBrothers2[i + 1]) <= 0);
+        }
+    }
+
+    @Test
+    public void test_getParsedBrothers_with_more_than_10_brothers() throws HSMBlockchainBookkeepingRelatedException {
+        BlockHeader blockHeader1 = blockHeaderBuilder.setNumber(1).build();
+        BlockHeader blockHeader2 = blockHeaderBuilder.setNumber(2).build();
+        BlockHeader blockHeader3 = blockHeaderBuilder.setNumber(3).build();
+
+        List<BlockHeader> block1Brothers = Arrays.asList(
+            blockHeaderBuilder.setNumber(101).build(),
+            blockHeaderBuilder.setNumber(102).build(),
+            blockHeaderBuilder.setNumber(103).build(),
+            blockHeaderBuilder.setNumber(104).build(),
+            blockHeaderBuilder.setNumber(105).build(),
+            blockHeaderBuilder.setNumber(106).build(),
+            blockHeaderBuilder.setNumber(107).build(),
+            blockHeaderBuilder.setNumber(108).build(),
+            blockHeaderBuilder.setNumber(109).build(),
+            blockHeaderBuilder.setNumber(110).build(),
+            blockHeaderBuilder.setNumber(111).build(),
+            blockHeaderBuilder.setNumber(112).build(),
+            blockHeaderBuilder.setNumber(113).build()
+        );
+
+        List<BlockHeader> block2Brothers = Arrays.asList(
+            blockHeaderBuilder.setNumber(201).build(),
+            blockHeaderBuilder.setNumber(202).build(),
+            blockHeaderBuilder.setNumber(203).build(),
+            blockHeaderBuilder.setNumber(204).build()
+        );
+
+        blocks = Arrays.asList(
+            new Block(blockHeader1, Collections.emptyList(), block1Brothers, true, true),
+            new Block(blockHeader2, Collections.emptyList(), block2Brothers, true, true),
+            new Block(blockHeader3, Collections.emptyList(), Collections.emptyList(), true, true)
+        );
+
+        AdvanceBlockchainMessage message = new AdvanceBlockchainMessage(blocks);
+        List<String> parsedBlockHeaders = message.getParsedBlockHeaders();
+
+        int brothersLimitPerBlockHeader = 10;
+        String[] brothers1 = message.getParsedBrothers(parsedBlockHeaders.get(2));
+        assertEquals(brothersLimitPerBlockHeader, brothers1.length);
+        assertNotEquals(blocks.get(0).getUncleList().size(), brothers1.length);
+
+        // filter top 10 from block1Brothers by difficulty value
+        String[] expectedBlock1Brothers = message.filterBrothers(block1Brothers).stream()
```

Comment:
You are basically comparing the result `message.filterBrothers` with the result of the same function called inside AdvanceBlockchainMessage class. It's like comparing `fun(a) == fun(a)`.

When creating the brothers you should set a difficulty to each of them, then you can know which are the ones that should be selected

Reply:
Updated